packages/database/src/core/util/ServerValues.ts

```diff
@@ -67,13 +116,19 @@ export const resolveDeferredValue = function(
  */
 export const resolveDeferredValueTree = function(
   tree: SparseSnapshotTree,
+  syncTree: SyncTree,
   serverValues: Object
 ): SparseSnapshotTree {
   const resolvedTree = new SparseSnapshotTree();
   tree.forEachTree(new Path(''), function(path, node) {
+    const existing = syncTree.calcCompleteEventCache(path);
```

Comment:
@inlined - Would it be possible to move this call into `resolveComplexDeferredValue`?

Context https://github.com/firebase/firebase-js-sdk/issues/2487